### PR TITLE
Add 'import-boss': enforce go import restrictions.

### PR DESCRIPTION
**Pull request by lavalamp**
Addresses #16768

Also fixes a few bugs in go2idl (package names weren't set properly).

### CODE REVIEW

**Comment by lavalamp** [comment]
@smarterclayton @nikhiljindal

File: cmd/libs/go2idl/import-restrictions [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "Rules": [
+    {
+      "SelectorRegexp": "k8s[.]io",
```
Comment (smarterclayton) [inline cmd/libs/go2idl/import-restrictions]: Does this belong in this package?

Also, would recommend this be a dot file

Comment (smarterclayton) [inline cmd/libs/go2idl/import-restrictions]: Nm, was non obvious at first this was controlling the tools itself

Comment (smarterclayton) [inline cmd/libs/go2idl/import-restrictions]: Should be a dot file still

**Comment by smarterclayton** [comment]
Looks fine to me - we can iterate more later

File: cmd/libs/go2idl/import-restrictions (hunk 1 above)
Comment (lavalamp) [inline cmd/libs/go2idl/import-restrictions]: ".import-restrictions"? That's fine, I was thinking it should be more visible, but I don't feel too strongly.

**Comment by smarterclayton** [comment]
I don't have a strong preference, just thinking about other similar mechanisms.

**Comment by lavalamp** [comment]
Filename changed. PTAL

**Comment by smarterclayton** [comment]
Lgtm - we can resolve any problems through use

**Comment by eparis** [comment]
#k8s-bot test this issue: #IGNORE

**Comment by eparis** [comment]
@k8s-bot test this issue: #IGNORE

**Pull request merged**